python/tvm/driver/tvmc/autotuner.py

```diff
@@ -255,97 +388,113 @@ def drive_tune(args):
         runner_ctor = (
-            auto_scheduler.LocalRunner if args.enable_autoscheduler else autotvm.LocalRunner
+            auto_scheduler.LocalRPCMeasureContext if enable_autoscheduler else autotvm.LocalRunner
         )
-        runner = runner_ctor(
-            number=args.number,
-            repeat=args.repeat,
-            timeout=args.timeout,
+        local_server = runner_ctor(
+            number=number,
+            repeat=repeat,
+            timeout=timeout,
             min_repeat_ms=min_repeat_ms,
         )
 
-    if args.enable_autoscheduler:
-        # Specify hardware parameters
-        hardware_params = auto_scheduler.HardwareParams(
-            args.num_cores,
-            args.vector_unit_bytes,
-            args.cache_line_bytes,
-            args.max_shared_memory_per_block,
-            args.max_local_memory_per_block,
-            args.max_threads_per_block,
-            args.max_vthread_extent,
-            args.warp_size,
-        )
+        # For autoscheduling on some devices, we need to maintain a LocalRPCMeasureContext object.
+        if enable_autoscheduler:
+            runner = local_server.runner
+        else:
+            runner = local_server
+
+    if enable_autoscheduler:
+
         tasks, weights = autoscheduler_get_tuning_tasks(
             mod=mod,
             params=params,
             target=target,
-            alter_layout=args.desired_layout,
+            alter_layout=desired_layout,
             hardware_params=hardware_params,
-            include_simple_tasks=args.include_simple_tasks,
+            include_simple_tasks=include_simple_tasks,
         )
 
+        # If not specified, choose a number of trials likely to produce good results.
+        if trials is None:
+            trials = 10000
+
         # Create the autoscheduler tuning options
         tuning_options = auto_scheduler.TuningOptions(
-            num_measure_trials=args.trials,
-            measure_callbacks=[auto_scheduler.RecordToFile(args.output)],
+            num_measure_trials=trials,
+            measure_callbacks=[auto_scheduler.RecordToFile(tuning_records)],
             runner=runner,
-            early_stopping=args.early_stopping,
+            early_stopping=early_stopping,
         )
 
         # Schedule the tasks (i.e., produce a schedule for each task)
-        schedule_tasks(
-            tasks, weights, tuning_options, args.tuning_records, args.log_estimated_latency
-        )
+        schedule_tasks(tasks, weights, tuning_options, prior_records, log_estimated_latency)
     else:
         tasks = autotvm_get_tuning_tasks(
             mod=mod,
             params=params,
             target=target,
-            alter_layout=args.desired_layout,
+            alter_layout=desired_layout,
         )
 
+        # If trails isn't specified, default to a number likely to produce good
+        # results without taking too much time.
+        if trials is None:
+            trials = 1000
```

Comment:
Yes I agree that unifying them in tvmc is ideal. Do you have a preference for total trials vs per-task trials? I lean towards total trials being more intuitive.

Reply:
I think it depends whether the user understands what task extraction does. if not, I think trials-per-task is intuitive. I understand that total trials better implies the runtime, but it's hard to estimate that if you don't know how task extraction behaves.

perhaps we need a tvmc subcommand which just runs and prints task extraction.

Reply:
If we assume users have limited understanding to TVM, then it they may don't care and don't know that tasks are as you pointed out. However, accordingly, trials per model is more intuitive to me. If users specify N trials, then they could just expect the total tuning time is `N * latency-per-trial` instead of `N * ?`.

Meanwhile, I agree that a subcommand to just print task extraction is a good idea. Both AutoTVM and AutoScheduler users can use this command to calculate the total trials.

Reply:
I've made the behavior for both autotvm and autotuner total number of trials and added some logging messages to indicate this.